enterprise/cypher/cypher/src/test/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/RegisteredPipeBuilderTest.scala

```diff
@@ -449,4 +449,29 @@ class RegisteredPipeBuilderTest extends CypherFunSuite with LogicalPlanningTestS
         PipelineInformation(Map("n" ->  LongSlot(0, nullable = false, CTNode, "n")),1, 0))())
   }
 
+  test("Should use NodeIndexUniqeSeek"){
+    // given
+    val lhs = NodeByLabelScan(x, LabelName("label")(pos), Set.empty)(solved)
+    val label = LabelToken("label2", LabelId(0))
+    val seekExpression = SingleQueryExpression(literalInt(42))
+    val rhs = NodeUniqueIndexSeek(z, label, Seq.empty, seekExpression, Set(x))(solved)
+    val apply = Apply(lhs, rhs)(solved)
```

Comment:
Do we really need to test unique index and 'index nested loop join' (ie. Apply with both lhs and rhs)? We could test the index alone.